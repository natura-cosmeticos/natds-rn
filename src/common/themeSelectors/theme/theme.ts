import { tokens as styleTokens } from '@naturacosmeticos/natds-styles';
import { IColorThemeTokens } from '@naturacosmeticos/natds-styles/dist/tokens/colors/themeBound/IColorThemeTokens';
import { IElevation } from '@naturacosmeticos/natds-styles/dist/tokens/elevation/IElevation';

type Tokens = typeof styleTokens;
type TokensWithoutColor = Omit<Tokens, 'colors'>
type TokensWithoutColorTokens = Omit<TokensWithoutColor, 'colorTokens'>
type TokensWithoutElevation = Omit<TokensWithoutColorTokens, 'elevation'>

interface ElevationProps {
  elevation: number
  shadowColor: string
  shadowOffset: { height: number, width: number }
  shadowOpacity: number
  shadowRadius: number
}

type Elevation = Record<keyof IElevation, ElevationProps>


export type Theme = TokensWithoutElevation
  & {
    colorTokens: IColorThemeTokens,
    elevation: Elevation | {}
  }

type ThemeMode = 'dark' | 'light';

const parseElevation = (elevation: string) => {
  const regEx = /(\d+(\.\d+)?)/g;
  const matches = elevation.match(regEx);

  if (matches) {
    const [width, height, blurRadius, , r, g, b, alpha] = matches;

    const toUnit = (prop: string) => parseInt(prop, 0);

    return {
      elevation: 2,
      shadowColor: `rgb(${r}, ${g}, ${b})`,
      shadowOffset: { height: toUnit(height), width: toUnit(width) },
      shadowOpacity: parseFloat(alpha),
      shadowRadius: toUnit(blurRadius),
    };
  }

  return {};
};

const buidElevation = (elevation: IElevation) => {
  const nativeElevation = {};

  Object
    .keys(elevation)
    .forEach((key) => { nativeElevation[key] = parseElevation(elevation[key]); });

  return nativeElevation;
};

export const buildTheme = (tokens: Tokens, brand: string, mode: ThemeMode) => {
  const colorThemeName = `${brand}${mode === 'light' ? 'Light' : 'Dark'}ColorTokens`;
  const localTokens = { ...tokens };

  delete localTokens.colors;

  const theme: Theme = {
    ...localTokens,
    colorTokens: localTokens.colorTokens[colorThemeName],
    elevation: buidElevation(localTokens.elevation),
  };

  return theme;
};

interface ThemeSelector {
  (theme: Theme): string | undefined
}

export const getTheme = (selector: ThemeSelector) => (
  { theme }: { theme: Theme },
) => selector(theme);
