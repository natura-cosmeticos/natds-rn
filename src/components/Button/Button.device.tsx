import { storiesOf } from '@storybook/react-native';
import {
  all,
  contained,
  disabled,
  interactive,
  outlined,
  text,
} from './Button.stories';

storiesOf('Button', module)
  .add('all', all)
  .add('contained', contained)
  .add('outlined', outlined)
  .add('text', text)
  .add('disabled', disabled)
  .add('interactive', interactive);
