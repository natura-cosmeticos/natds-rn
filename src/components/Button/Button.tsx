import React from 'react';
import styled, { withTheme } from 'styled-components/native';
import { NativeSyntheticEvent, NativeTouchEvent } from 'react-native';
import {
  getColorPrimary,
  getColorOnPrimary,
  getColorHighEmphasis,
  getButtonPropsBySize,
  getFont,
  getRadiusBySize,
  getShadowBySize,
  Theme,
  getOpacity10,
  getColorPrimaryLight,
} from '../../common/themeSelectors';

export type ButtonTypes = 'contained' | 'outlined' | 'text'

export interface ButtonProps {
  /**
   * The onPress event handler
   */
  onPress: (ev: NativeSyntheticEvent<NativeTouchEvent>) => void,
  /**
   * The button text content
   */
  text: string
  /**
   * Button variants `contained` | `outlined` | `text`
   */
  type?: ButtonTypes
  /**
   * The button theme
   */
  theme: Theme,
  /**
   * An accessibility hint helps users understand what will happen when they perform an action
   * on the accessibility element when that result is not clear from the accessibility label.
   */
  accessibilityHint?: string
  /**
   * Overrides the text that's read by the screen reader when the user interacts with the element.
   * By default, the label is constructed by traversing all the children and accumulating
   * all the Text nodes separated by space.
   */
  accessibilityLabel?: string
   /**
   * Optional ID for testing
   */
  testID?: string,
}

interface ButonBase {
  type: ButtonTypes
  theme: Theme
}

const isContained = (type: ButtonTypes) => type === 'contained';

const getButtonStyles = (theme: Theme, type: ButtonTypes) => {
  const styles = {
    contained: {
      background: getColorPrimary(theme),
    },
    outlined: {
      borderColor: getColorPrimary(theme),
      borderWidth: 1,
    },
  };

  return styles[type];
};

const getButtonTextStyles = (theme: Theme, type: ButtonTypes) => (
  isContained(type)
    ? getColorOnPrimary(theme)
    : getColorHighEmphasis(theme)
);

const ButtonBase = styled.TouchableHighlight<ButonBase>(({ type, theme }) => ({
  borderRadius: getRadiusBySize(theme, 'medium'),
  ...getButtonStyles(theme, type),
  ...getButtonPropsBySize(theme, 'medium'),
}));

const Text = styled.Text<ButonBase>`
  color: ${({ type, theme }) => getButtonTextStyles(theme, type)};
  font-size: 14px;
  align-self: center;
  font-weight: 600;
  letter-spacing: 1px;
  font-family: ${getFont('bold')};
`;

const getShadowByType = (type: ButtonTypes, theme: Theme) => (
  isContained(type)
    ? getShadowBySize(theme, '2')
    : {}
);

const ButtonComponent = ({
  onPress, theme, text, type = 'contained', testID = 'button', accessibilityLabel, accessibilityHint,
}: ButtonProps) => (
  <ButtonBase
    testID={testID}
    type={type}
    onPress={onPress}
    style={getShadowByType(type, theme)}
    underlayColor={getColorPrimaryLight(theme)}
    activeOpacity={getOpacity10(theme)}
  >
    <Text
      accessibilityLabel={accessibilityLabel}
      accessibilityHint={accessibilityHint}
      accessibilityRole="button"
      type={type}
    >{text.toUpperCase()}</Text>
  </ButtonBase>
);

export const Button = withTheme(ButtonComponent);
