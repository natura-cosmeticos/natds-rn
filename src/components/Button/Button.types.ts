import { IconName } from '@naturacosmeticos/natds-icons';
import { Color, Theme } from '@naturacosmeticos/natds-themes/react-native';

/**
 * @deprecated Use 'semi' and 'medium'
 */
type ButtonSizesDeprecated = 'small' | 'large'
export type ButtonSizes = 'semi' | 'semiX' | 'medium' | ButtonSizesDeprecated

export type ButtonTypes = 'contained' | 'outlined' | 'text'
export type IconPositions = 'left' | 'right'

export interface ButtonBaseProps extends ButtonProps {
  textColor: keyof Color
  theme: Theme
}

export interface ButtonProps {
  /**
   * An accessibility hint helps users understand what will happen when they perform an action
   * on the accessibility element when that result is not clear from the accessibility label.
   */
  accessibilityHint?: string
  /**
   * Overrides the text that's read by the screen reader when the user interacts with the element.
   * By default, the label is constructed by traversing all the children and accumulating
   * all the Text nodes separated by space.
   */
  accessibilityLabel?: string
  /**
   * A disabled button is unusable and un-clickable.
   * The disabled attribute can be set to keep a user from clicking on the button until some
   * other condition has been met (like selecting a checkbox, etc.).
   */
  disabled?: boolean
  /**
   * The icon name
   */
  iconName?: IconName,
  /**
   * Icon position relative to text label `left` | `right`
   */
  iconPosition?: IconPositions
  /**
   * The onPress event handler
   */
  onPress: () => void,
  /**
  * Optional ID for testing
  */
  testID?: string,
  /**
   * Icon Button sizes `semi` | `semiX` | `medium`
   *
   * #### deprecated sizes: `small` and `large`:
   * - use `semi` and `medium` instead
   */
  size?: ButtonSizes,
  /**
   * The button text content
   */
  text: string
  /**
   * Button variants `contained` | `outlined` | `text`
   */
  type?: ButtonTypes
}
