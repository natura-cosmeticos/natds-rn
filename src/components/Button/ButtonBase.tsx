/* eslint-disable max-lines */
import React from 'react';
import styled, { withTheme } from 'styled-components/native';
import { NativeSyntheticEvent, NativeTouchEvent } from 'react-native';
import {
  IColors,
  Theme,
  getButtonPropsBySize,
  getColorHighEmphasis,
  getColorLowEmphasis,
  getColorMediumEmphasis,
  getColorOnPrimary,
  getColorPrimary,
  getColorPrimaryLight,
  getOpacity10,
  getRadiusBySize,
  getShadowBySize,
  getColorByName,
} from '../../common/themeSelectors';

export type ButtonTypes = 'contained' | 'outlined' | 'text'
export type TextColors = keyof IColors | 'default'

export interface ButtonProps {
  accessibilityHint?: string
  accessibilityLabel?: string
  disabled?: boolean
  textColor: TextColors
  onPress: (ev: NativeSyntheticEvent<NativeTouchEvent>) => void,
  testID?: string
  text: string
  theme: Theme
  type?: ButtonTypes
}

const isContained = (type: ButtonTypes) => type === 'contained';

const getButtonStyles = (theme: Theme, type: ButtonTypes, disabled: boolean) => {
  const styles = {
    contained: {
      background: disabled ? getColorLowEmphasis(theme) : getColorPrimary(theme),
    },
    outlined: {
      borderColor: disabled ? getColorMediumEmphasis(theme) : getColorPrimary(theme),
      borderWidth: 1,
    },
  };

  return styles[type];
};

const getButtonTextColor = (theme: Theme, type: ButtonTypes, disabled: boolean) => {
  const color = {
    active: isContained(type) ? getColorOnPrimary(theme) : getColorHighEmphasis(theme),
    disabled: isContained(type) ? getColorHighEmphasis(theme) : getColorMediumEmphasis(theme),
  };

  return disabled ? color.disabled : color.active;
};

const getShadowByType = (type: ButtonTypes, disabled: boolean, theme: Theme) => (
  isContained(type) && !disabled
    ? getShadowBySize(theme, '2')
    : {}
);

const Base = styled.TouchableHighlight<{
  type: ButtonTypes
  disabled: boolean
  theme: Theme
}>(({
  type, theme, disabled = false,
}) => ({
  ...getButtonPropsBySize(theme, 'medium'),
  ...getButtonStyles(theme, type, disabled),
  borderRadius: getRadiusBySize(theme, 'medium'),
}));

const Label = styled.Text<{
  type: ButtonTypes
  disabled: boolean
  textColor: TextColors
  theme: Theme
}>(({
  type, theme, disabled, textColor,
}) => ({
  alignSelf: 'center',
  color: textColor === 'default' ? getButtonTextColor(theme, type, disabled) : getColorByName(theme, textColor),
  fontSize: 14,
  fontWeight: 'bold',
  letterSpacing: 1,
}));

const ButtonComponent = ({
  accessibilityHint,
  accessibilityLabel,
  disabled = false,
  textColor,
  onPress,
  testID = 'button',
  text,
  theme,
  type = 'contained',
}: ButtonProps) => (
  <Base
    activeOpacity={getOpacity10(theme)}
    disabled={disabled}
    onPress={disabled ? () => {} : onPress}
    style={getShadowByType(type, disabled, theme)}
    testID={testID}
    type={type}
    underlayColor={getColorPrimaryLight(theme)}
  >
    <Label
      accessibilityHint={accessibilityHint}
      accessibilityLabel={accessibilityLabel}
      accessibilityRole="button"
      disabled={disabled}
      textColor={textColor}
      type={type}
    >{text.toUpperCase()}</Label>
  </Base>
);

export const ButtonBase = withTheme(ButtonComponent);
