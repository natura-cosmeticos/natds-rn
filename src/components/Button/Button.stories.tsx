
import React from 'react';
import { View } from 'react-native';
import { text as textKnob, select } from '@storybook/addon-knobs';
import { ContainerRow, ContainerWithTheme } from '../../common/HelperComponents/ThemeHelper.styles';
import { Button, ButtonTypes } from './Button';

export default {
  component: Button,
  parameters: {
    componentSubtitle: 'Buttons allow users to take actions, and make choices, with a single tap.',
  },
  title: 'Components|Button',
};

const onPress = () => { };

const buttonTypes = {
  contained: 'contained',
  outlined: 'outlined',
  text: 'text',
};

export const all = () => (
  <View style={{ maxWidth: 600, padding: 30 }}>
    <ContainerRow style={{ marginBottom: 30 }}>
      <Button onPress={onPress} text="default" />
      <Button onPress={onPress} text="default" type="outlined" />
      <Button onPress={onPress} text="default" type="text" />
    </ContainerRow >
    <ContainerWithTheme style={{ height: 200, justifyContent: 'space-between' }} >
      <Button onPress={onPress} text="default" />
      <Button onPress={onPress} text="default" type="outlined" />
      <Button onPress={onPress} text="default" type="text" />
    </ContainerWithTheme >
  </View>
);

export const contained = () => (
  <ContainerRow style={{ padding: 30 }} >
    <Button onPress={onPress} text="default" type="contained" />
  </ContainerRow>
);

export const outlined = () => (
  <ContainerRow style={{ padding: 30 }}>
    <Button onPress={onPress} text="default" type="outlined" />
  </ContainerRow>
);

export const text = () => (
  <ContainerRow style={{ padding: 30 }}>
    <Button onPress={onPress} text="default" type="text" />
  </ContainerRow>
);

export const interactive = () => (
  <ContainerRow style={{ padding: 30 }} >
    <Button
      onPress={onPress}
      text={textKnob('Text', 'default')}
      type={select('Types', buttonTypes, 'contained') as ButtonTypes}
    />
  </ContainerRow>
);
