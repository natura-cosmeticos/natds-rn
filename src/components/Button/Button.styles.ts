/* eslint-disable max-lines */
import { Color } from '@naturacosmeticos/natds-themes/react-native';
import styled from 'styled-components/native';
import {
  getColorByName,
  getColorLowEmphasis,
  getColorMediumEmphasis,
  getColorPrimary,
  getRadiusBySize,
  getShadowBySize,
  getSizeMedium,
  getSizeSemi,
  getSizeSemiX,
  getSpacingMicro,
  getSpacingSmall,
  getSpacingTiny,
} from '../../common/themeSelectors';
import { ButtonBaseProps } from './Button.types';

type SurfaceProps = Required<Pick<ButtonBaseProps, 'type' | 'theme' | 'disabled' | 'size'>>
type LabelProps = Pick<ButtonBaseProps, 'iconName' | 'iconPosition' | 'textColor' | 'theme' | 'disabled'| 'type'>

const getButtonStylesByType = ({ disabled, theme, type = 'contained' }: Omit<SurfaceProps, 'size'>) => {
  const styles = {
    contained: {
      background: disabled ? getColorLowEmphasis(theme) : getColorPrimary(theme),
    },
    outlined: {
      borderColor: disabled ? getColorMediumEmphasis(theme) : getColorPrimary(theme),
      borderWidth: 1,
    },
  };

  return styles[type];
};

export const getButtonStylesBySize = ({ size, theme }: Pick<SurfaceProps, 'size' | 'theme'>) => {
  const buttonSizes = {
    large: {
      minHeight: getSizeMedium(theme),
      paddingHorizontal: getSpacingSmall(theme),
    },
    medium: {
      minHeight: getSizeMedium(theme),
      paddingHorizontal: getSpacingSmall(theme),
    },
    semiX: {
      minHeight: getSizeSemiX(theme),
      paddingHorizontal: getSpacingTiny(theme),
    },
    // eslint-disable-next-line sort-keys
    semi: {
      minHeight: getSizeSemi(theme),
      paddingHorizontal: getSpacingMicro(theme),
    },
    small: {
      minHeight: getSizeSemi(theme),
      paddingHorizontal: getSpacingMicro(theme),
    },
  };

  return buttonSizes[size];
};

export const getButtonShadowByType = ({ disabled, theme, type }: Omit<SurfaceProps, 'size'>) => (
  type === 'contained' && !disabled
    ? getShadowBySize(theme, 'tiny')
    : {}
);

export const getButtonTextColor = ({ disabled, type }: Pick<LabelProps, 'disabled'| 'type'>) => {
  const color: {
    active: keyof Color,
    disabled: keyof Color,
  } = {
    active: type === 'contained' ? 'onPrimary' : 'highEmphasis',
    disabled: type === 'contained' ? 'highEmphasis' : 'mediumEmphasis',
  };

  return disabled ? color.disabled : color.active;
};

export const Surface = styled.View<SurfaceProps>(({
  type = 'contained', theme, disabled = false, size,
}) => ({
  ...getButtonStylesBySize({ size, theme }),
  ...getButtonStylesByType({ disabled, theme, type }),
  alignContent: 'center',
  alignItems: 'center',
  borderRadius: getRadiusBySize(theme, 'medium'),
  justifyContent: 'center',
}));

const iconMargin = ({ theme, iconName, iconPosition }: Pick<LabelProps, 'theme' | 'iconName' | 'iconPosition'>) => {
  if (!iconName) {
    return {};
  }

  return iconPosition === 'left'
    ? { marginStart: getSpacingTiny(theme) }
    : { marginEnd: getSpacingTiny(theme) };
};

const labelOrder = ({ iconPosition }: Pick<LabelProps, 'iconPosition'>) => (
  iconPosition === 'right' ? 'row' : 'row-reverse'
);

export const Label = styled.View<Pick<LabelProps, 'iconPosition'>>(({ iconPosition }) => (
  { alignItems: 'center', flexDirection: labelOrder({ iconPosition }) }
));

export const LabelText = styled.Text<LabelProps>(({
  iconName, iconPosition, textColor, theme,
}) => ({
  ...iconMargin({ iconName, iconPosition, theme }),
  color: getColorByName(theme, textColor),
  fontWeight: theme.typography.fontWeight.medium,
  lineHeight: theme.typography.lineHeight.medium * theme.body2.fontSize,
}));
