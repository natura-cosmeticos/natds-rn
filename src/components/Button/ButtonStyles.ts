import styled from 'styled-components/native';
import {
  getButtonPropsBySize,
  getColorLowEmphasis,
  getColorMediumEmphasis,
  getColorPrimary,
  getRadiusBySize,
  getShadowBySize,
  getColorByName,
  getSpacingTiny,
} from '../../common/themeSelectors';
import { ButtonBaseProps } from './Button.types';

const getButtonStylesByType = ({ disabled, theme, type = 'contained' }: Omit<SurfaceProps, 'size'>) => {
  const styles = {
    contained: {
      background: disabled ? getColorLowEmphasis(theme) : getColorPrimary(theme),
    },
    outlined: {
      borderColor: disabled ? getColorMediumEmphasis(theme) : getColorPrimary(theme),
      borderWidth: 1,
    },
  };

  return styles[type];
};

const getButtonShadowByType = ({ disabled, theme, type }: Omit<SurfaceProps, 'size'>) => (
  type === 'contained' && !disabled
    ? getShadowBySize(theme, 'tiny')
    : {}
);

type SurfaceProps = Required<Pick<ButtonBaseProps, 'type' | 'theme' | 'disabled' | 'size'>>

export const Surface = styled.View<SurfaceProps>(({
  type = 'contained', theme, disabled = false, size,
}) => ({
  ...getButtonPropsBySize(theme, size),
  ...getButtonStylesByType({ disabled, theme, type }),
  ...getButtonShadowByType({ disabled, theme, type }),
  alignContent: 'center',
  alignItems: 'center',
  borderRadius: getRadiusBySize(theme, 'medium'),
  justifyContent: 'center',
}));

const iconMargin = ({ theme, iconName, iconPosition }: Omit<LabelProps, 'textColor'>) => {
  if (!iconName) {
    return {};
  }

  return iconPosition === 'left'
    ? { marginStart: getSpacingTiny(theme) }
    : { marginEnd: getSpacingTiny(theme) };
};

const labelOrder = ({ iconPosition }: Pick<LabelProps, 'iconPosition'>) => (
  iconPosition === 'right' ? 'row' : 'row-reverse'
);

type LabelProps = Pick<ButtonBaseProps, 'iconName' | 'iconPosition' | 'textColor' | 'theme'>

export const Label = styled.View<Pick<LabelProps, 'iconPosition'>>(({ iconPosition }) => (
  { alignItems: 'center', flexDirection: labelOrder({ iconPosition }) }
));

export const LabelText = styled.Text<LabelProps>(({
  iconName, iconPosition, textColor, theme,
}) => ({
  ...iconMargin({ iconName, iconPosition, theme }),
  color: getColorByName(theme, textColor),
  fontWeight: theme.typography.fontWeight.medium,
  lineHeight: theme.typography.lineHeight.medium * theme.body2.fontSize,
}));
