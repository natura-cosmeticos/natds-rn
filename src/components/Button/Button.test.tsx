import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react-native';
import '@testing-library/jest-native/extend-expect';
import { ThemeProvider } from 'styled-components/native';
import theme from '../../common/themeSelectors/theme/mock-theme.json';
import { Button, ButtonProps } from './Button';

jest.mock('react-native/Libraries/Components/Touchable/TouchableHighlight',
  () => 'TouchableHighlight');

jest.mock('../../common/themeSelectors', () => (
  {
    getButtonPropsBySize: () => ({ background: '#AEAEAE' }),
    getColorHighEmphasis: () => '#FAF3E3',
    getColorOnPrimary: () => '#F4F4',
    getColorPrimary: () => '#FFFFFF',
    getColorPrimaryLight: () => '#BABABA',
    getFont: () => 'doc robot',
    getOpacity80: () => 0.8,
    getRadiusBySize: () => 42,
    getShadowBySize: () => ({ shadowColor: '#AEAEAE' }),
  }));

const renderButton = (fn, props: Omit<ButtonProps, 'theme'>) => (fn(
  <ThemeProvider theme={theme}>
    <Button {...props} />
  </ThemeProvider>,
));

const defaultProps = ({
  onPress: () => {},
  text: 'label button',
});

describe('Button component', () => {
  it('Should render button with default props', () => {
    const { queryByTestId } = renderButton(render, defaultProps);

    expect(queryByTestId('button')?.props).toHaveProperty('type', 'contained');
  });

  it('Should render button with a uppercase text', () => {
    const { queryByTestId } = renderButton(render, defaultProps);

    expect(queryByTestId('button')).toHaveTextContent('LABEL BUTTON');
  });

  it('Should render button with the given type prop', () => {
    const { queryByTestId } = renderButton(render, {
      ...defaultProps,
      type: 'outlined',
    });

    expect(queryByTestId('button')).toHaveStyle({
      borderWidth: 1,
    });
  });

  it('Should call the given onPress function', () => {
    const onPressMock = jest.fn();
    const { queryByTestId } = renderButton(render, {
      ...defaultProps,
      onPress: onPressMock,
    });

    fireEvent.press(queryByTestId('button'));

    expect(onPressMock).toHaveBeenCalledTimes(1);
  });

  it('Should render button component outlined', () => {
    const button = renderButton(renderer.create, {
      ...defaultProps,
      type: 'outlined',
    }).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('Should render button component contained', () => {
    const button = renderButton(renderer.create, defaultProps).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('Should render button component text', () => {
    const button = renderButton(renderer.create, {
      ...defaultProps,
      type: 'text',
    }).toJSON();

    expect(button).toMatchSnapshot();
  });
});
