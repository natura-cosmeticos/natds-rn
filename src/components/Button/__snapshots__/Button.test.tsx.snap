// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Should render button component contained 1`] = `
<TouchableHighlight
  activeOpacity={0.8}
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#AEAEAE",
        "borderRadius": 42,
      },
      Object {
        "shadowColor": "#AEAEAE",
      },
    ]
  }
  testID="button"
  type="contained"
  underlayColor="#BABABA"
>
  <Text
    accessibilityRole="button"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#F4F4",
          "fontFamily": "doc robot",
          "fontSize": 14,
          "fontWeight": "600",
          "letterSpacing": 1,
        },
      ]
    }
    type="contained"
  >
    LABEL BUTTON
  </Text>
</TouchableHighlight>
`;

exports[`Button component Should render button component outlined 1`] = `
<TouchableHighlight
  activeOpacity={0.8}
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#AEAEAE",
        "borderColor": "#FFFFFF",
        "borderRadius": 42,
        "borderWidth": 1,
      },
      Object {},
    ]
  }
  testID="button"
  type="outlined"
  underlayColor="#BABABA"
>
  <Text
    accessibilityRole="button"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#FAF3E3",
          "fontFamily": "doc robot",
          "fontSize": 14,
          "fontWeight": "600",
          "letterSpacing": 1,
        },
      ]
    }
    type="outlined"
  >
    LABEL BUTTON
  </Text>
</TouchableHighlight>
`;

exports[`Button component Should render button component text 1`] = `
<TouchableHighlight
  activeOpacity={0.8}
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#AEAEAE",
        "borderRadius": 42,
      },
      Object {},
    ]
  }
  testID="button"
  type="text"
  underlayColor="#BABABA"
>
  <Text
    accessibilityRole="button"
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#FAF3E3",
          "fontFamily": "doc robot",
          "fontSize": 14,
          "fontWeight": "600",
          "letterSpacing": 1,
        },
      ]
    }
    type="text"
  >
    LABEL BUTTON
  </Text>
</TouchableHighlight>
`;
